### ISSUE TITLE
Incorrect type inference of "Extended Iterable Unpacking"

### ISSUE THREAD

**Issue by jacobbogdanov** [microsoft/pyright#758]
**Describe the bug**
unpacking a sequence seems to infer the wrong type of the "star expression". It should be a `List`, but `pyright` infers an `Iterable`.

**To Reproduce**
Run `pyright` against the code example below.

**Expected behavior**
My reading of [Pep 3132](https://www.python.org/dev/peps/pep-3132/) is that the when unpacking an iterable, the `*rest` always turns into a `list`, regardless of the type of the expression.
> This designates a subexpression that will be **assigned a list of all items** from the iterable being unpacked that are not assigned to any of the mandatory expressions, or an empty list if there are no such items.
(emphasis mine)

**Screenshots or Code**
Here's a contrived example,

```python
from typing import Callable, Sequence

def for_each(items: Sequence[str], func: Callable[[str], None]) -> None:
    first, *rest = items  # rest is deduced as Iterable[str]
    func(first)
    # assert isinstance(rest, list)  # un-commenting this makes the error go away
    for_each(rest, func)
           # ^^^^ error
```
produces the following error:
```
  7:14 - error: Argument of type "Iterable[str]" cannot be assigned to parameter "items" in function "for_each" of type "Sequence[str]"
  "Iterable[str]" is incompatible with "Sequence[str]" (reportGeneralTypeIssues)
1 error, 0 warnings 
```

**VS Code extension or command-line**
Pyright via VSCode extention v1.1.45

**Comment by erictraut** [comment]
Thanks for the bug report. You are correct, it should be a List, not an Iterable. I misread the spec when I originally implemented it. This will be fixed in the next published version of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.46, which I just published.
